File: frontend/contractor-babu-app/src/types/site.ts
// Site Types
export interface Site {
    id: string
    name: string
    address: string
    city?: string | null
    state?: string | null
    zipCode?: string | null
    clientId: string
    clientName?: string
    createdAt: string
    createdBy?: string
    updatedAt?: string | null
    updatedBy?: string | null
    isActive: boolean
}

export interface CreateSiteDto {
    name: string
    address: string
    city?: string
    state?: string
    zipCode?: string
    clientId: string
}

export interface UpdateSiteDto {
    name?: string
    address?: string
    city?: string
    state?: string
    zipCode?: string
    isActive?: boolean
}

File: backend/CleanArchitectureApp.Domain/Entities/Site.cs
using CleanArchitectureApp.Domain.Entities;

namespace CleanArchitectureApp.Domain.Entities;

public class Site : TenantEntity
{
    public string Id { get; set; } = string.Empty;
    public required string Name { get; set; }
    public required string Address { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? ZipCode { get; set; }
    
    // Relationships
    public string ClientId { get; set; } = string.Empty;
    public Client? Client { get; set; }
    
    public ICollection<Project> Projects { get; set; } = new List<Project>();
    public ICollection<Quotation> Quotations { get; set; } = new List<Quotation>();
    
    // Audit
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string CreatedBy { get; set; } = string.Empty;
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }
    public bool IsActive { get; set; } = true;
}
